src/brake/kia_soul_ev_niro/brake_module.rs

```diff
@@ -84,38 +85,55 @@ impl UnpreparedBrakeModule {
 }
 
 impl BrakeModule {
-    fn disable_control(&mut self, debug_console: &mut DebugConsole) {
+    pub fn disable_control(&mut self, debug_console: &mut DebugConsole) -> Result<(), OxccError> {
+        let mut result = Ok(());
+
         if self.control_state.enabled {
             self.brake_pedal_position.prevent_signal_discontinuity();
 
-            self.brake_dac.output_ab(
+            if let Err(_) = self.brake_dac.output_ab(
                 DacOutput::clamp(self.brake_pedal_position.low()),
                 DacOutput::clamp(self.brake_pedal_position.high()),
-            );
+            ) {
+                result = Err(OxccError::IO);
+            }
 
+            // even if we've encountered an error, we can still disable
             self.brake_pins.spoof_enable.set_low();
             self.brake_pins.brake_light_enable.set_low();
             self.control_state.enabled = false;
             writeln!(debug_console, "Brake control disabled");
         }
+
+        result
     }
 
-    fn enable_control(&mut self, debug_console: &mut DebugConsole) {
+    fn enable_control(&mut self, debug_console: &mut DebugConsole) -> Result<(), OxccError> {
+        let mut result = Ok(());
```

Comment:
It is usually more Rustic to construct a result at the point of decision/return rather than creating and mutating a result output variable. The fact that if/else branches return values that can in turn be automatically returned from functions if they are the final statement makes this easier.